examples/model_selection/plot_nested_cross_validation_iris.py

```diff
@@ -0,0 +1,119 @@
+"""
+=========================================
+Nested versus Non-Nested cross-validation
+=========================================
+
+This example compares non-nested and nested cross-validation strategies on a
+classifier of the iris data set. Nested cross-validation (CV) is often used to
+train a model in which hyperparameters also need to be optimized. Nested CV
+approximates the generalization error of the resulting estimator of the
+hyperparameter search. It is generally best practice to use nested CV as
+non-nested CV will sometimes provide a slightly more biased and optimistic
+score.
+
+In contrast to non-nested CV, an inner CV loop is introduced that partitions
+the training set from the outer loop into two additional sets, a training set
+and validation set. We thus end up with a train/validation/test data split in
+nested CV versus the traditional train/test data split in non-nested CV. The
+reason is to avoid testing the model on data that we have also used to tune the
+model parameters because information could "leak" into our model and overfit
+the data. This magnitude of this effect is primarily dependent on the size of
+the dataset and the stability of the model. For more quantitative detail of
+potential bias when tuning parameters, see this paper.  [1]_
+
+Each iteration of the inner CV loop will provide the best estimator for the
```

Comment:
dropped paragraph.
It was a point of confusion for me when first starting so I thought I would point it out explicitly, but i'm not at all attached.